src/categorical_algebra/FinSets.jl

```diff
@@ -104,12 +104,29 @@ function product(A::FinOrd, B::FinOrd)
   Span(π1, π2)
 end
 
+function product(Xs::Vector{FinOrd{T}}) where {T}
```

Comment:
Is it worth reducing all the binary products, coproducts, etc to the general n-ary case, to avoid redundant code? The only reason not to do it would be performance, but I suspect that the difference will be negligible.

Reply:
Also, do `product` and `coproduct` work in the special case `length(Xs) == 0`, where they should give terminal and initial objects, respectively?

Reply:
I think that the question about the binary coproducts depend on whether you want to use (Co)Span as a datastructure. It seems to me like there are cases where we want to use (Co)Span, and not (Co)Cone, so it might make sense to keep around the special case.

Also, something that I thought of is that if we care about efficiency, then, for instance, in a limit, any object in the diagram which has an arrow pointing to it does not need to be in the big product that is then filtered, because its value is always determined by some other thing. If we implemented this optimization, then the performance of the general case would be more similar to the performance of the specialized case in pushouts/pullbacks. Maybe that would be good for another PR though, as that would complicate the implementation.

Reply:
OK, then let's keep as is for now. Even if you follow up with a more complicated but more efficient implementation for the general case, it might be good to keep the simple special case just for clarity.

Reply:
I'm worried about the performance implications of using n-ary instead of binary, because it not only involves the for-loop overhead, but also the memory allocation. If these were tuples it wouldn't matter because julia can use stack allocation for tuples, but with Vector{T} it heap allocates. If we relaxed the type constraint here, you could pass a (f,g)::Tuple{FinOrd, FinOrd} and get the best of both worlds. In that case we should just assume that `Xs` is iterable and use that interface for both binary and n-ary (co)products